pkg/daemon/daemon.go

```diff
@@ -742,3 +666,58 @@ func ValidPath(path string) bool {
 	}
 	return false
 }
+
+// SenseAndLoadOnceFrom gets a hold of the content for supported onceFrom configurations,
+// parses to verify the type, and returns back the genericInterface, the type description,
+// if it was local or remote, and error.
+func (dn *Daemon) SenseAndLoadOnceFrom() (interface{}, string, string, error) {
+	var content []byte
+	var err error
+	var contentFrom string
+	// Read the content from a remote endpoint if requested
+	if strings.HasPrefix(dn.onceFrom, "http://") || strings.HasPrefix(dn.onceFrom, "https://") {
+		contentFrom = MachineConfigOnceFromRemoteConfig
+		resp, err := http.Get(dn.onceFrom)
+		if err != nil {
+			return nil, "", contentFrom, err
+		}
+		defer resp.Body.Close()
+		// Read the body content from the request
+		content, err = dn.fileSystemClient.ReadAll(resp.Body)
+		if err != nil {
+			return nil, "", contentFrom, err
+		}
+	} else {
+		// Otherwise read it from a local file
+		contentFrom = MachineConfigOnceFromLocalConfig
+		absoluteOnceFrom, err := filepath.Abs(filepath.Clean(dn.onceFrom))
+		if err != nil {
+			return nil, "", contentFrom, err
+		}
+
+		content, err = dn.fileSystemClient.ReadFile(absoluteOnceFrom)
+		if err != nil {
+			return nil, "", contentFrom, err
+		}
+	}
+
+	// Try each supported parser
+	// TODO: Add to resource/machineconfig.go as a function?
+	mcfgScheme := runtime.NewScheme()
+	mcfgv1.AddToScheme(mcfgScheme)
+	mcfgCodecs := serializer.NewCodecFactory(mcfgScheme)
+	requiredObj, err := runtime.Decode(mcfgCodecs.UniversalDecoder(mcfgv1.SchemeGroupVersion), content)
+	if err == nil {
+		glog.V(2).Infof("onceFrom file is of type MachineConfig")
+		mcConfig := requiredObj.(*mcfgv1.MachineConfig)
+		return mcConfig, MachineConfigMCFileType, contentFrom, nil
+	}
+	glog.V(2).Infof("%s is not a MachineConfig: %s. Trying Ignition.", dn.onceFrom, err)
+
+	ignConfig, _, err := ignv2.Parse(content)
```

Comment:
Possibly this should be tried first to avoid the scaffolding required for the MC case?

Reply:
I like that idea. I'll switch it up.